### ISSUE TITLE
knife bootstrap "Node   exists, overwrite it? (Y/N)"

### ISSUE THREAD

**Issue by jblaine** [chef/chef#3558]
Chef 12.3.0

```
% knife bootstrap sprint-rh6.our.org --bootstrap-proxy http://gate.our.org:80 --sudo --ssh-user jblaine -i ~/Private/id_rsa_dmz
Node  exists, overwrite it? (Y/N)
```

**Comment by thommay** [comment]
Was there more to this report?

**Comment by jblaine** [comment]
No. I indicated the host to bootstrap and the command reported an incomplete message. Note the spaces:

```
Node  exists, overwrite it? (Y/N)
```

**Comment by lamont-granquist** [comment]
-N is mandatory to give a node name.  12.4.0 fixes this so you don't get an edge case you get an error condition.

**Comment by ranjib** [comment]
Yup, this is happening because node name is evalued to empty string, hence the double space. @jblaine can you test this on 12.4 ? it should be fixed

**Comment by lamont-granquist** [comment]
This could also be reopened as a new ticket for "validatorless bootstraps need to support pulling the node name from the target host's hostname" since there's a known gap in functionality there.

**Comment by jblaine** [comment]
Seems fixed in 12.4.x

    ERROR: You must pass a node name with -N when bootstrapping with user credentials
